src/methods/SANA.cpp

```diff
@@ -1835,20 +1835,52 @@ uint SANA::getHighestIndex() const {
     return highestIndex;
 }
 
-#define LOG10_LOW_TEMP -10.0
-#define LOG10_HIGH_TEMP 10.0
-#define LOG10_NUM_STEPS 20 // looping through 0 <= i <= LOG10_NUM_STEPS
+#define PBAD_HIGH_TEMP_LIMIT 0.99999
+#define PBAD_LOW_TEMP_LIMIT 1e-10
+
+
+double SANA::temperatureBracket(double LIMIT){
+	int i = 0;
+	while (true){
+		double pBadi = pForTInitial(pow(10,i));
+		double pBadi1 = pForTInitial(pow(10,i+1));
+		cout << "Testing 10 ^ " << i << " pBad = " << pBadi << endl;
+		if (pBadi < LIMIT && pBadi1 >= LIMIT){
+			break;
+		}else if (pBadi > LIMIT){
+			i--;
+		}
+		else if(pBadi < LIMIT){
+			i++;
+		}
+	}
+	return LIMIT == PBAD_HIGH_TEMP_LIMIT ? i+1 : i;
```

Comment:
Please add a second parameter to the call to this function specifying if this is a high or a low temperature, and use that Boolean to make this decision. Don't assume any particular value for the LIMIT.